packages/cdktf-cli/lib/dependencies/package-manager.ts

```diff
@@ -0,0 +1,217 @@
+import { Language } from "@cdktf/provider-generator";
+import { existsSync } from "fs-extra";
+import path from "path";
+import { exec } from "../util";
+import { xml2js, js2xml, Element } from "xml-js";
+import { Errors } from "../errors";
+import * as fs from "fs-extra";
+
+/**
+ * manages installing, updating, and removing dependencies
+ * in the package system used by the target language of a CDKTF
+ * project
+ */
+export abstract class PackageManager {
+  constructor(protected readonly workingDirectory: string) {}
+
+  public static forLanguage(
+    language: Language,
+    workingDirectory: string
+  ): PackageManager {
+    switch (language) {
+      case Language.GO:
+        throw new Error("Go is not supported yet");
+      case Language.TYPESCRIPT:
+        return new NodePackageManager(workingDirectory);
+      case Language.PYTHON:
+        return new PythonPackageManager(workingDirectory);
+      case Language.CSHARP:
+        return new NugetPackageManager(workingDirectory);
+      case Language.JAVA:
+        return new MavenPackageManager(workingDirectory);
+      default:
+        throw new Error(`Unknown language: ${language}`);
+    }
+  }
+
+  public abstract addPackage(
+    packageName: string,
+    packageVersion?: string
+  ): Promise<void>;
+  // public abstract listPackages(): Promise<todo>; future stuff..
```

Comment:
Nit: Should these stay?

Reply:
would be fine with me, but I have no strong opinion at all here :)

Reply:
me neither